test/integration/scheduler/framework_test.go

```diff
@@ -581,8 +581,8 @@ func TestUnreservePlugin(t *testing.T) {
 				if err = waitForPodUnschedulable(cs, pod); err != nil {
 					t.Errorf("test #%v: Didn't expected the pod to be scheduled. error: %v", i, err)
 				}
-				if unresPlugin.numUnreserveCalled == 0 || unresPlugin.numUnreserveCalled != pbdPlugin.numPrebindCalled {
-					t.Errorf("test #%v: Expected the unreserve plugin to be called %d times, was called %d times.", i, pbdPlugin.numPrebindCalled, unresPlugin.numUnreserveCalled)
+				if unresPlugin.numUnreserveCalled < 1 {
```

Comment:
I don't think this is the right fix. 

The problem is that the tests in this file modify some shared state (the members in TesterPlugin above). What we need to do is reset this state for every test. As a quick fix, you can reset pbdPlugin.numPrebindCalled and unresPlugin.numUnreserveCalled to zero at the beginning of the for loop. 

But what we should really be doing is refactor this whole file to ensure that each test starts from a clean state to make sure that this doesn't happen again in the future.

Reply:
It does the reset here https://github.com/kubernetes/kubernetes/pull/79371/files#diff-f6b72dfe2b2c2cee201d12156e80366fL596 and the tests don't run in parallel...

I agree with the proposed refactoring, I still feel like this is the right fix at this point :)

Reply:
> It does the reset here https://github.com/kubernetes/kubernetes/pull/79371/files#diff-f6b72dfe2b2c2cee201d12156e80366fL596 

reset() should be called at the beginning, not the end.

> and the tests don't run in parallel...

The tests don't run in parallel, but they may run in different order. 

> 
> I agree with the proposed refactoring, I still feel like this is the right fix at this point :)

I don't think it is, we should expect bind and unreserve plugins to be executed the same number of times.

Reply:
The problem is that TestPrebindPlugin does not reset the counter. But a test shouldn't be relying on another, so the right fix is for this test to make sure the state it relies on is reset before it runs.

Reply:
on a closer look, I think you are right that we can't guarantee that they are equal because the scheduler and the test are running in parallel; what we can guarantee is that prebind counter is at most off by one :)

I still think that we have a bigger problem that the tests modify shared state and rely on each other to make sure the state is reset. 

For now, in addition to the modification you did, I would suggest that you also reset the prebind counter in TestPrebindPlugin.

Reply:
@ahg-g what about the race I described? I saw that in logs...

we wait for pod being unschedulable
that happens, the pod goes back to scheduling queue
then we check (at later time) that those two numbers (prebind, unreserve) equal...
but in the meantime, the pod got rescheduled and the prebind was executed, but not the unreserve one (YET)

Reply:
oh you replied in the meantime, good 👍 

sure I can do that! And I can work on the refactoring after if no one else is working on that

Reply:
@ahg-g added the reset to the other test. You can re-review